### PR TITLE
add and remove from server

### PR DESCRIPTION
**Pull request by vladyslav-harkusha**
[DEMO LINK](https://vladyslav-harkusha.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

**Review by xiimiik — APPROVED** [review]
GJ

File: src/App.tsx [hunk 1]
```diff
@@ -1,223 +1,199 @@
...
+      }
+    } catch (error) {
+      setIsError(true);
+      setErrorText('Unable to load todos');
+
+      throw new Error(`unexpected error with loading todos: ${error}`);
+    } finally {
+      setTimeout(() => {
+        setIsError(false);
+      }, 3000);
```
Comment (Andr1yK) [inline src/App.tsx:48]: Mb, we need to do it only if we have an error :)

File: src/components/FilterForTodos/FilterForTodos.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={cn('filter__link',
+            { selected: filterBy === TodoStatus.ALL })}
+          onClick={() => setFilterBy(TodoStatus.ALL)}
+        >
+          All
+        </a>
```
Comment (Andr1yK) [inline src/components/FilterForTodos/FilterForTodos.tsx:40]: let's try render this items from array

File: src/components/NewTodo/NewTodo.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, { useCallback, useState } from 'react';
+
+type Props = {
+  newTodoField: React.RefObject<HTMLInputElement>;
+  addNewTodo: (todoTitle: string) => Promise<void>;
+  isTodoAdding: boolean;
+};
+
+export const NewTodo: React.FC<Props> = React.memo(({
+  newTodoField, addNewTodo, isTodoAdding,
```
Comment (Andr1yK) [inline src/components/NewTodo/NewTodo.tsx:10]:
```suggestion
  newTodoField,
  addNewTodo,
  isTodoAdding,
```

File: src/components/TodoItem/TodoItem.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+}) => {
+  const [deleteButtonClicked, setDeleteButtonClicked] = useState(false);
+  const { title, completed, id } = todo;
+
+  const handleDelete = () => {
+    setDeleteButtonClicked(true);
+    removeTodo(id);
+  };
+
+  const loaderIsActive = () => {
```
Comment (Andr1yK) [inline src/components/TodoItem/TodoItem.tsx:23]: why is it fn?

File: src/components/TodoItem/TodoItem.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    if (id === 0 || deleteButtonClicked) {
+      return true;
+    }
+
+    if (isDeletingCompleted && completed) {
+      return true;
+    }
+
+    return false;
```
Comment (Andr1yK) [inline src/components/TodoItem/TodoItem.tsx:32]:
```suggestion
    return (id === 0 || deleteButtonClicked)
      || (isDeletingCompleted && completed);
```

File: src/components/TodosList/TodosList.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+type Props = {
+  todos: Todo[];
+  removeTodo: (todoId: number) => void;
+  isTodoAdding: boolean;
+  tempTodo: Todo;
+  isDeletingCompleted: boolean;
+};
+
+export const TodosList: React.FC<Props> = React.memo(({
+  todos, removeTodo, isTodoAdding, tempTodo, isDeletingCompleted,
```
Comment (Andr1yK) [inline src/components/TodosList/TodosList.tsx:16]: Why do u write this inline?

**Review by Andr1yK — CHANGES_REQUESTED** [review]

**Review by Andr1yK — APPROVED** [review]

**Pull request closed without merging**